src/Microsoft.ML.Sweeper/Algorithms/SmacSweeper.cs

```diff
@@ -163,20 +163,20 @@ private FastForestRegressionPredictor FitModel(IEnumerable<IRunResult> previousR
         /// <returns>An array of ParamaterSets which are the candidate configurations to sweep.</returns>
         private ParameterSet[] GenerateCandidateConfigurations(int numOfCandidates, IEnumerable<IRunResult> previousRuns, FastForestRegressionPredictor forest)
         {
-            ParameterSet[] configs = new ParameterSet[numOfCandidates];
```

Comment:
I don't see how this change is related to making things internal. Can you explain?

Reply:
So, as explained elsewhere, making things internal does not change the objects by itself, but using a different object (MLContext instead of ConsoleEnvironment) might, since they are distinct. In this case as we've seen elsewhere this can use the random state differently.

Now then, as we see above, the methods this called *can* randomly generate fewer configurations than were asked for -- but this condition was not being hit in the test. But if I change the seed a little, or the way in which the PRNG is initialized in some other way, then suddenly I see I'm having an exception thrown.

Lo and behold, I find a bug that the mechanism was not uncovering, but which does suddenly become apparent as soon as the new mechanism is used. So I fix the bug.